core/test/Test/Pos/Core/Arbitrary.hs

```diff
@@ -253,6 +254,9 @@ instance Arbitrary AddrStakeDistribution where
                         CoinPortion <$> choose (1, max 1 (limit - 1))
                     genPortions (n - 1) (portion : res)
 
+instance Arbitrary NetworkMagic where
+    arbitrary = oneof [pure NMNothing, NMJust <$> arbitrary]
```

Comment:
This will use the `arbitrary` of `Int32` which permits negative values. Is that what we want here?

Reply:
The type should probably be `Word32`.

Reply:
`ProtocolMagic` uses Int32, which I agree is a bad choice and should be `Word32`. I'm fairly sure that `fromIntegral` will do the conversion we want, but I also worry that we could be bitten by mixing `Word32`/`Int32` types in `ProtocolMagic` and `NetworkMagic`.